app/scripts/services/Message.js

```diff
@@ -0,0 +1,19 @@
+(function() {
+    function Message($firebaseArray) {
+        var ref = firebase.database().ref().child('messages');
+        var messages = $firebaseArray(ref);
+        
+        
+        return {
+            getByRoomId: function (roomId) {
+                var query = ref.orderByChild('roomId').equalTo(roomId);
+                var currentList = $firebaseArray(query);
```

Comment:
Note you could chain this all together.. like `return $firebaseArray(ref.orderByChild('roomId').equalTo(roomId))`.   Which way do you prefer? Any reason why?